src/java/org/apache/cassandra/repair/AccordRepairJob.java

```diff
@@ -91,30 +94,35 @@ protected void runRepair()
     @Override
     void abort(@Nullable Throwable reason)
     {
-        throw new UnsupportedOperationException("Have not implemented this yet, and the job runs synchronously so it isn't abortable");
+        shouldAbort = reason == null ? new RuntimeException("Abort") : reason;
     }
 
-    private void repairRange(TokenRange range)
+    private void repairRange(TokenRange range) throws Throwable
     {
         RoutingKey remainingStart = range.start();
         BigInteger rangeSize = splitter.sizeOf(range);
         if (rangeStep == null)
-            rangeStep = BigInteger.ONE.max(splitter.divide(rangeSize, 1000));
+        {
+            BigInteger divide = splitter.divide(rangeSize, 1000);
+            rangeStep = divide.equals(BigInteger.ZERO) ? rangeSize : BigInteger.ONE.max(divide);
+        }
 
         BigInteger offset = BigInteger.ZERO;
 
         TokenRange lastRepaired = null;
         int iteration = 0;
         while (true)
         {
+            if (shouldAbort != null)
+                throw shouldAbort;
             iteration++;
-            if (iteration % 100 == 0)
+            if (iteration % ACCORD_REPAIR_RANGE_STEP_UPDATE_INTERVAL.getInt() == 0)
```

Comment:
can you cache `ACCORD_REPAIR_RANGE_STEP_UPDATE_INTERVAL` so we don't do this every time in the loop?